Classes/YLRefreshHeaderView/YLRefreshHeaderView.m

```diff
@@ -31,26 +35,37 @@ - (void)setRefreshState:(YLRefreshHeaderViewState)refreshState {
 }
 
 - (void)setRefreshState:(YLRefreshHeaderViewState)refreshState animated:(BOOL)animated {
-  _refreshState = refreshState;
+  if (_refreshState != refreshState) {
+    _refreshState = refreshState;
 
-  void (^animationBlock)() = ^{
-    UIScrollView *strongScrollView = self.scrollView;
-    UIEdgeInsets contentInset = strongScrollView.contentInset;
-    // TODO(mglidden): make this support view controllers that have UIRectEdgeTop
-    contentInset.top = (refreshState == YLRefreshHeaderViewStateRefreshing) ? self.pullAmountToRefresh : 0;
-    strongScrollView.contentInset = contentInset;
-  };
-  void (^completionBlock)(BOOL) = ^(BOOL finished) {
-    if (self.refreshState == YLRefreshHeaderViewStateClosing) {
-      self.refreshState = YLRefreshHeaderViewStateNormal;
+    void (^animationBlock)() = ^{
+      UIScrollView *strongScrollView = self.scrollView;
+      UIEdgeInsets contentInset = strongScrollView.contentInset;
+      switch (refreshState) {
+        case YLRefreshHeaderViewStateRefreshing:
+          self.previousTopInset = contentInset.top;
+          contentInset.top = self.pullAmountToRefresh;
```

Comment:
So, while trying out the branch I saw that if you have UIRectEdgeTop enabled, your contentInset.top should be something like 64. However, if my pullAmountToRefresh is 15 for instance, I would swap this 64 with 15 which would mess around the view. What I meant is that it probably needs 64 + the pullAmountToRefresh, so basically:
contentInset.top = self.previousTopInset + self.pullAmountToRefresh.

To test this correctly you can open the YLTableViewExample workspace and change UIRectEdgeNone to UIRectEdgeTop on L31 of YLExampleViewController. You'll see that if you use this branch as is it will not produce the expected behavior